packages/vendure-plugin-webhook/src/api/webhook.service.ts

```diff
@@ -177,34 +213,67 @@ export class WebhookService implements OnApplicationBootstrap {
    * Call the actual webhook with the configured Transformer for given Event
    */
   async callWebhook(webhook: Webhook, event: EventWithContext): Promise<void> {
+    let headers: HeadersInit = {};
+
+    // Add default authorization header if configured
+    if (this.options.useAuthorizationHeader) {
+      const token = await this.webhookToken.value();
+      if (token) {
+        headers['authorization'] = token;
+      } else {
+        Logger.error(
+          `Could not get webhook token but 'useAuthorizationHeader' is enabled.`,
+          loggerCtx
+        );
+      }
+    }
```

Comment:
No need to do that. I consider the self refreshing cache to be reliable. If the token is not defined, it's only because it's intentional or because of a missing database migration.